projects/angular/src/forms/datepicker/date-input.ts

```diff
@@ -299,4 +301,13 @@ export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implement
       .pipe(filter(date => !!date))
       .subscribe(() => this.datepickerFocusService.focusInput(this.el.nativeElement));
   }
+
+  private retriggerValidation() {
+    if (this.datepickerHasFormControl() && !this.usingNativeDatepicker()) {
+      this.control?.control?.updateValueAndValidity({ onlySelf: true, emitEvent: false }); // Added onlySelf, emitEvent purposefully inorder to prevent unnecessary valuechange triggers
```

Comment:
I'm concerned about not emitting `statusChanges` here because it can change from valid to invalid or vice versa. I would rather emit extra events than skip events and possibly cause hard-to-find bugs. Consumers of `valueChanges` and `statusChanges` can use `distinctUntilChanged` if needed.

Reply:
Agree! statusChange event will not be triggered when emitEvent is set to false, but control setErrors method will trigger the validations and its already triggering the statusChange event. Removing emitEvent option would trigger additional valueChanges at application end on every min/max setters.

Reply:
Ah, got it. Thanks!